src/readfish/plugins/_filter.py

```diff
@@ -76,6 +166,40 @@ def __call__(self, obj):
 
 
 class Filters:
+    """
+    Represents a collection of filter conditions parsed from a list of rule strings.
+
+    This class aggregates multiple `Filter` objects and provides a unified interface
+    to evaluate a set of filter conditions against given objects. It evaluates whether
+    an object satisfies *all* the provided conditions.
+
+    :param rules: A list of filtering rule strings to parse and interpret.
+    :type rules: List[str]
+
+    :Example:
+
+    .. doctest::
+
+        >>> rules = ["is_primary", "mapq > 40", "strand == -1"]
+        >>> filters = Filters(rules)
+        >>> obj1 = {"is_primary": True, "mapq": 50, "strand": -1}
+        >>> obj2 = {"is_primary": False, "mapq": 50, "strand": -1}
+        >>> obj3 = {"is_primary": True, "mapq": 30, "strand": -1}
+        >>> filters(obj1)
+        True
+        >>> filters(obj2)
+        False
+        >>> filters(obj3)
+        False
+
```

Comment:
Good to see this.

Presumably we should also make sure that any filters are evaluated and reported in validate?

Reply:
Yes, good idea!

Reply:
It would be useful to have some kind of report string that is "human readable" to provide the user...

we should always remember Uncle Bens advice...

Reply:
"It's rice time"